### PR TITLE
rpc server: make possible to disable/enable batch requests

### PR DESCRIPTION
**Pull request by niklasad1**
The rationale behind this, is that it may be useful for some users actually disable RPC batch requests or limit them by length instead of the total size bytes of the batch.

This PR adds two new CLI options:

```
--rpc-disable-batch-requests - disable batch requests on the server
--rpc-max-batch-request-len <LEN> - limit batches to LEN on the server.
```

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5240627

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5241066

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5241643

File: substrate/client/cli/src/commands/run_cmd.rs [hunk 1]
```diff
@@ -399,6 +407,22 @@ impl CliConfiguration for RunCmd {
 		Ok(self.rpc_max_subscriptions_per_connection)
 	}
 
+	fn rpc_buffer_capacity_per_connection(&self) -> Result<u32> {
```
Comment (niklasad1) [inline substrate/client/cli/src/commands/run_cmd.rs:410]: I noticed that this was missing :(

File: prdoc/pr_3364.prdoc [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# See doc at https://raw.githubusercontent.com/paritytech/polkadot-sdk/master/prdoc/schema_user.json
+
+title: rpc server expose batch request configuration
+
+doc:
+  - audience: Node Operator
+    description: |
+      Add functionality to limit RPC batch requests by two new CLI options:
+      --rpc-disable-batch-request - disable batch requests on the server
+      --rpc_max_batch_request_len - limit batches to LEN on the server
```
Comment (jsdw) [inline prdoc/pr_3364.prdoc:11]: Why is one `-` and one `_`?

Comment (niklasad1) [reply prdoc/pr_3364.prdoc:11]: oops my bad, copy-pasted from the code

Comment (niklasad1) [inline prdoc/pr_3364.prdoc:11]:
```suggestion
      --rpc-max-batch-request-len - limit batches to LEN on the server
```

File: substrate/client/cli/src/commands/run_cmd.rs (hunk 1 above)
Comment (jsdw) [reply substrate/client/cli/src/commands/run_cmd.rs:422]: Ooh this is the line I missed when I was looking at the PR and only noticed the batch config bits :)

**Review by jsdw — APPROVED** [review]

**Review by skunert — APPROVED** [review]

File: .gitlab/pipeline/check.yml [hunk 3]
```diff
@@ -133,6 +133,7 @@ check-runtime-migration-westend:
     WASM: "westend_runtime.compact.compressed.wasm"
     URI: "wss://westend-try-runtime-node.parity-chains.parity.io:443"
     SUBCOMMAND_EXTRA_ARGS: "--no-weight-warnings"
+  allow_failure: true
```
Comment (niklasad1) [inline .gitlab/pipeline/check.yml:136]: CI fix to allow the job to fail

**Pull request merged**